### ISSUE TITLE
Cannot fetch depth information on Vulkan backend (only on mac platform)

### ISSUE THREAD

**Issue by Morcki** [taichi-dev/taichi#5954]
**To Reproduce**

```py
# sample code here
import taichi as ti

ti.init(arch=ti.vulkan) # Correct one is to use arch=ti.cuda

def test():
    window = ti.ui.Window("test", (512, 512), vsync=True, show_window=False)
    canvas = window.get_canvas()
    scene = ti.ui.Scene()
    camera = ti.ui.Camera()

    scene.set_camera(camera)
    canvas.scene(scene)
        
    window.get_depth_buffer_as_numpy()
    window.destroy()

test()
```

**Log/Screenshots**
Failed to run the python script and report error information as below, but it run correctly on cuda backend.

```
[Taichi] version 1.1.3, llvm 10.0.0, commit ef35996b, osx, python 3.10.6
[Taichi] Starting on arch=vulkan
[I 09/02/22 11:37:35.613 33994301] [vulkan_device_creator.cpp:pick_physical_device@372] Found Vulkan Device 0 (Apple M1)
[I 09/02/22 11:37:35.614 33994301] [vulkan_device_creator.cpp:create_logical_device@440] Vulkan Device "Apple M1" supports Vulkan 0 version 1.1.216
[W 09/02/22 11:37:35.614 33994301] [vulkan_device_creator.cpp:create_logical_device@488] Potential non-conformant Vulkan implementation, enabling VK_KHR_portability_subset
[W 09/02/22 11:37:35.614 33994301] [vulkan_device_creator.cpp:create_logical_device@556] Taichi GPU GUI requires wide lines support
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: vkAllocateMemory(): External memory requires a dedicated VkBuffer or VkImage.
/Users/longyuhao/WorkSpace/Dev/taichi/python/taichi/_kernels.py:204: UserWarning: Calling non-taichi function "len". Scope inside the function is not processed by the Taichi AST transformer. The function may not work as expected. Proceed with caution! Maybe you can consider turning it into a @ti.func?
  static_assert(len(normal_arr.shape) == 2)
[E 09/02/22 11:37:35.689 33994301] [spirv_ir_builder.cpp:get_primitive_type@274] Type f64 not supported.
```

**Comment by bobcao3** [comment]
Why is there fp64 involved? It's probably an accidental mistake, we should avoid using double precision as much as possible in general

**Comment by Morcki** [comment]
I think so, and this bug only report on mac.

**Comment by bobcao3** [comment]
> I think so, and this bug only report on mac.

That's because so far in our testing platform only mac does not support FP64. On desktop FP64 is supported but we should still avoid this due to performance reasons. Need to find our where it went into FP64 and correct it to use FP32

**Comment by Morcki** [comment]
Yes, I will figure out this problem.

**Comment by Morcki** [comment]
I see, numpy default type is f64... And cannot be passed to taichi kernel on mac. Fixed this on #5947 .
And I think maybe we should add a warning to mac users when they pass a f64 type arg into kernel.

**Comment by bobcao3** [comment]
> I see, numpy default type is f64... And cannot be passed to taichi kernel on mac. Fixed this on #5947 . And I think maybe we should add a warning to mac users when they pass a f64 type arg into kernel when choosing vulkan backend.

I think fp64 also doesn't work on Metal.
